src/viewer.ts

```diff
@@ -370,62 +372,122 @@ class Viewer {
 
     private initXrMode() {
         const backup = {
-            position: new Vec3()
+            position: new Vec3(),
+            rotation: new Vec3()
```

Comment:
I normally think of a rotation as a Quat and naming like `eulers` if you store Euler angles. Not a big deal tho.